src/main/java/org/opentripplanner/middleware/triptracker/TravelerLocator.java

```diff
@@ -91,6 +101,46 @@ public static String getInstruction(
         return NO_INSTRUCTION;
     }
 
+    public static void checkForLegTransition(TripStatus tripStatus, TravelerPosition travelerPosition, MonitoredTrip trip) {
+        if (
+            hasRequiredTripStatus(tripStatus) &&
+            (hasRequiredWalkLeg(travelerPosition) || hasRequiredTransitLeg(travelerPosition))
+        ) {
+            List<NotificationType> legTransitionTypes = getLegTransitionTypes(travelerPosition);
+            if (!legTransitionTypes.isEmpty()) {
+                try {
+                    new CheckMonitoredTrip(trip).processLegTransition(legTransitionTypes, travelerPosition);
+                } catch (CloneNotSupportedException e) {
+                    LOG.error("Error encountered while checking leg transition.", e);
+                }
+            }
+        }
+
+    }
+
+    private static List<NotificationType> getLegTransitionTypes(TravelerPosition travelerPosition) {
+        List<NotificationType> notificationTypes = new ArrayList<>();
+        if (isAtStartOfLeg(travelerPosition)) {
+            notificationTypes.add(DEPARTED_NOTIFICATION);
+        }
+        if (isAtEndOfLeg(travelerPosition)) {
```

Comment:
If arriving from a bus leg with the bus shape in the middle of the road, and the stop is away from the curb, I think the threshold might not be met. Not sure if we want to use the "upcoming" radius instead of the "immediate" radius.

Reply:
Updated in: https://github.com/ibi-group/otp-middleware/pull/275/commits/91d66190cb611283993b8e745148641efd695c5e